canosp2020/preprocessing.py

```diff
@@ -71,9 +81,57 @@ def preprocess(self, text) -> spacy.tokens.doc.Doc:
             print(e)
             return None
 
+    @staticmethod
+    def get_stop_words(input_file="data/tickets_word2vec.model", eval=lambda x: x <= 0.02) -> List[str]:
+        """
+        Get a list of step words base on relative frequency.
+        The input could either be the raw CSV file or word2vec model build with genism.
+
+        The input format will be determined by the input_file extension <filename>.[csv|model].
+
+        The `eval` method is a function which takes a float variable,
+        word frequency, as a single argument and return a boolean value
+        which represent whether a word is a stop word or not.
+        Default word with frequency lower than 0.02 is considered as stop word.
```

Comment:
What is the threshold of 0.02 based on? For full language modelling this is a reasonable value, but since our corpus is domain-constrained it may be worth considering a higher threshold.

Reply:
I make 0.02 up.

To clarify a bit, are we considering high-frequency words as stop words?

If that's the case, with threshold 0.02 will in fact eliminate all words but `firefox` from our corpus since the highest frequency is `0.038` (firefox), and the second-highest is 0.01-ish

Reply:
> What is the threshold of 0.02 based on? For full language modelling this is a reasonable value, but since our corpus is domain-constrained it may be worth considering a higher threshold.

To clarify, are we considering low-frequency words as stop words? Or is it high frequency?

Reply:
> > What is the threshold of 0.02 based on? For full language modelling this is a reasonable value, but since our corpus is domain-constrained it may be worth considering a higher threshold.
> 
> To clarify, are we considering low-frequency words as stop words? Or is it high frequency?

I think we consider high-frequency words to be the stop words. Because they are common across all the tickets, these words might not that important to our model.

Reply:
I agree with @konantian, we should remove words that have a high frequency in our text. Also 0.02 could be a bit small. There are a lot of words that are used very often, but are still meaningful. 
I applied TF-IDF to your preprocessed tickets and still found that words like "instead", "new" go through.